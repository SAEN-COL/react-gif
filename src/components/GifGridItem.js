import React from 'react'
import PropType from 'prop-types';
export const GifGridItem = ( { id, title, url } ) => {
  return (
    <div className="card animate__animated animate__fadeIn animate__delay-1s">
      <img src={ url } alt={ title } />
      <p> { title } </p>
    </div>
  )
};

GifGridItem.propTypes = {
  title: PropType.string.isRequired,
  url: PropType.string.isRequired,
}