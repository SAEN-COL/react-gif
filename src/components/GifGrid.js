
import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';


export const GifGrid = ( { category }) => {
  
  const { data:images, loading } = useFetchGifs( category );

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> { category } </h3>
      
      {loading && <p className="animate__animated animate__flash">Loadding</p> }
      <div className="card-grid">
          { images.map(
            image =>  
              <GifGridItem 
                key={ image.id } 
                {...image} />   // Los tres puntos es para indicar 
                                // que los key del objeto se pasan
                                // como parametros o props del componente
          )}
      </div>
    </>
  )
}
