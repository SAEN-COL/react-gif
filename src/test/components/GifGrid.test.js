import React from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Validación componente <GrifGrid />', () => {
  const category = 'love';
  
  test('Debería renderizar correctamente el componente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });    
    const wrapper  = shallow(<GifGrid category={ category } />);
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imágenes useFetchGifs (hook)', () => {    
    const gifs = [{
      id: 'ABC',
      url: 'https://example.com/img.jpg',
      title: 'Title image'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper  = shallow(<GifGrid category={ category } />);
    
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    
  });
  
  
});
