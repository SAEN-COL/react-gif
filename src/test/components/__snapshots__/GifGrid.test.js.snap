// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validación componente <GrifGrid /> Debe de mostrar items cuando se cargan imágenes useFetchGifs (hook) 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    love
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Title image"
      url="https://example.com/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`Validación componente <GrifGrid /> Debería renderizar correctamente el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    love
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loadding
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
